caddyhttp/basicauth/basicauth.go

```diff
@@ -60,6 +61,10 @@ func (a BasicAuth) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error
 
 			// remove credentials from request to avoid leaking upstream
 			r.Header.Del("Authorization")
+
+			// let upstream middleware (e.g. fastcgi and cgi) know about authenticated
+			// user; this replaces the request with a wrapped instance
+			r = r.WithContext(context.WithValue(r.Context(), "remote_user", username))
```

Comment:
@jung-kurt Okay, I've merged the other branch, so if you want to wrap your string key in httpserver.CtxKey, then I think we're good to go!